frontend/server/src/main/java/com/amazonaws/ml/mms/http/InferenceRequestHandler.java

```diff
@@ -152,12 +152,17 @@ private void predict(
             FullHttpRequest req,
             QueryStringDecoder decoder,
             String modelName)
-            throws ModelNotFoundException {
+            throws ModelNotFoundException, BadRequestException {
         RequestInput input = parseRequest(ctx, req, decoder);
         if (modelName == null) {
             modelName = input.getStringParameter("model_name");
-            if (modelName == null) {
-                throw new BadRequestException("Parameter model_name is required.");
+            if (modelName == null || modelName.isEmpty()) {
+                if (ModelManager.getInstance().getStartupModels().size() == 1) {
```

Comment:
We might not need this code. We already have this code above, in [handleInvocations()](https://github.com/awslabs/mxnet-model-server/blob/master/frontend/server/src/main/java/com/amazonaws/ml/mms/http/InferenceRequestHandler.java#L129). For context,
this is to support one of our customers endpoints. When /invocations is called, the customer doesn't provide a model name. Since we can potentially have multiple models, we had to come up with a intelligent way to figure out which model was being called. What MMS does is, when a model-name is not given, with /invocations, we will look for the model which is loaded at start-up of MMS. If there is no startup-time models or if there are multiple models loaded during start-up, we throw error. If there is only one model loaded at startup time, we will use that model for predict request. Since this is only for /invocations endpoint, we don't need to handle this in the generic "predict()" method. What do you say?

Reply:
agree